pxr/imaging/hdSt/materialXShaderGen.cpp

```diff
@@ -508,11 +508,9 @@ HdStMaterialXShaderGen<Base>::_EmitMxVertexDataDeclarations(
     }
 
     for (size_t i = 0; i < block.size(); ++i) {
-        line += _EmitMxVertexDataLine(block[i], separator);
-        // remove the separator from the last data line
-        if (i == block.size() - 1) {
-            line = line.substr(0, line.size() - separator.size());
-        }
+        auto const& lineSeparator =
+            (i == block.size() - 1) ? mx::EMPTY_STRING : separator;
+        line += _EmitMxVertexDataLine(block[i], lineSeparator);
```

Comment:
Found while testing some assets. Make sure we do not add the commas in the first place because there might be more than one to remove afterwards.